From bef7d8e1432400f3d78339ac269167e09c15dabd Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Sat, 20 Feb 2021 11:00:57 +0800
Subject: [PATCH] lockup_detector: init lockup detector after all the
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: bef7d8e1432400f3d78339ac269167e09c15dabd

 init_calls

hulk inclusion
category: feature
bugzilla: 48046
CVE: NA

-------------------------------------------------------------------------

We call 'sdei_init' as 'subsys_initcall_sync'. lockup detector need to
be initialised after sdei_init. The influence of this patch is that we
can not detect the hard lockup in init_calls.

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>

 Conflicts:
	init/main.c
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 init/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init/main.c b/init/main.c
index a9da7794333b..cff7e6cd990c 100644
--- a/init/main.c
+++ b/init/main.c
@@ -1516,7 +1516,6 @@ static noinline void __init kernel_init_freeable(void)
 
 	rcu_init_tasks_generic();
 	do_pre_smp_initcalls();
-	lockup_detector_init();
 
 	smp_init();
 	sched_init_smp();
@@ -1528,6 +1527,8 @@ static noinline void __init kernel_init_freeable(void)
 
 	do_basic_setup();
 
+	lockup_detector_init();
+
 	kunit_run_all_tests();
 
 	console_on_rootfs();
-- 
2.26.2

